src/sentry/auth/access.py

```diff
@@ -200,6 +207,26 @@ def from_request(request, organization=None, scopes=None):
     return from_user(request.user, organization, scopes=scopes)
 
 
+def from_sentry_app(user, organization=None):
+    if not organization:
+        return NoAccess()
+
+    sentry_app = SentryApp.objects.get(proxy_user=user)
+
+    if not sentry_app.is_installed_on(organization):
+        return NoAccess()
+
+    return Access(
+        scopes=sentry_app.scope_list,
```

Comment:
Instead of scopes coming from a token/User, it comes from the SentryApp. This will be the list of scopes they registered the application with. This list is static once created, for now, but we may change that in the future. Access being generated from this, instead of token scopes, means as they potentially change, they'll be reflected here without us needing to backfill token scopes.

Reply:
Do applications have a way to update/request more scopes?